server/src/core/system-lock/tests/system-lock.test.ts

```diff
@@ -16,91 +16,109 @@ describe('SystemLock tests', () => {
   });
 
-  beforeEach(async () => {
-    await dbUtils.resetCollectionWith(systemLocks);
-  });
+  // beforeEach(async () => {
+  //   await dbUtils.resetCollectionWith(systemLocks);
+  // }, 10000);
 
   describe('lock', () => {
+    beforeAll(async () => {
+      await dbUtils.resetCollectionWith(systemLocks);
+    }, 10000);
+
     test('should set locked:true on lock record', async () => {
       const lock = new SystemLockManager('lock2', dbUtils.getCollection());
       await lock.lock();
       const record = await dbUtils.getCollection().findOne({ _id: 'lock2' });
       expect(record.locked).toBe(true);
-    });
+    }, 10000);
 
     test('should create lock record with locked:true if record does not exist', async () => {
-      await dbUtils.dropCollection();
       const lock = new SystemLockManager('lock2', dbUtils.getCollection());
       await lock.lock();
       const record = await dbUtils.getCollection().findOne({ _id: 'lock2' });
       expect(record.locked).toBe(true);
-    });
+    }, 10000);
 
     test('should throw error if already locked', async () => {
-      const lock = new SystemLockManager('lock1', dbUtils.getCollection());
+      const lock = new SystemLockManager('lock2', dbUtils.getCollection());
       try {
         await lock.lock();
       }
       catch (e) {
         expect(e.code).toBe('systemLockLocked');
       }
-    });
+    }, 10000);
   });
 
   describe('unlock', () => {
+    beforeAll(async () => {
+      await dbUtils.resetCollectionWith(systemLocks);
+    }, 10000);
+
     test('should set locked to false if locked', async () => {
       const lock = new SystemLockManager('lock1', dbUtils.getCollection());
+      const records = await dbUtils.getCollection().find({ }).toArray();
+      console.log('In unlock records: ', records);
+      await lock.lock();
+      let record = await dbUtils.getCollection().findOne({ _id: 'lock1' });
+      console.log('after lock record: ', record);
+      expect(record.locked).toBe(true);
       await lock.unlock();
-      const record = await dbUtils.getCollection().findOne({ _id: 'lock1' });
+      record = await dbUtils.getCollection().findOne({ _id: 'lock1' });
+      console.log('after unlock record: ', record);
       expect(record.locked).toBe(false);
-    });
+    }, 10000);
     
-    test('should throw invalid state error if not locked', async () => {
+    test('should not set locked to false if not locked', async () => {
       const lock = new SystemLockManager('lock2', dbUtils.getCollection());
-      try {
-        await lock.unlock();
-      }
-      catch (e) {
-        expect(e.code).toBe('systemLockInvalidState');
-      }
-    });
+      let record = await dbUtils.getCollection().findOne({ _id: 'lock2' });
+      expect(record.locked).toBe(false);
+      await lock.unlock();
+      record = await dbUtils.getCollection().findOne({ _id: 'lock2' });
+      // console.log('record: ', record);
+      expect(record.locked).toBe(false);
+    }, 10000);
```

Comment:
This test description is not accurate. When a lock is not locked, then its state `locked:false`. So to say "should not set locked to false" while your test is asserting that `record.locked` is false at the end, doesn't make sense. Maybe rename it to "should not change lock state if already unlocked"